### PR TITLE
feat(eventlog): add event log UI

### PR DESCRIPTION
**Pull request by chethanbellechalu-intel**
Closes [AB#3610](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/3610), [AB#3611](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/3611), [AB#3614](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/3614)

### CODE REVIEW

File: src/app/devices/event-log/event-log.component.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+              item.eventTypeDesc = ''
+              break
+          }
+        })
+        this.dataSource.data = this.eventLogData
+      })
+    })
+  }
+
+  isNoData (): boolean {
```
Comment (rsdmike) [inline src/app/devices/event-log/event-log.component.ts:68]: this function doesn't have a unit test

File: src/app/devices/event-log/event-log.component.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          console.log(err)
+          this.snackBar.open($localize`Error retrieving event log`, undefined, SnackbarDefaults.defaultError)
+          return of(this.eventLogData)
+        }), finalize(() => {
+          this.isLoading = false
+        })
+      ).subscribe(data => {
+        this.eventLogData = data
+        this.eventLogData?.forEach(item => {
+          switch (item.EventType) {
```
Comment (rsdmike) [inline src/app/devices/event-log/event-log.component.ts:45]: none of these cases are tested with unit tests

**Review by rsdmike — CHANGES_REQUESTED** [review]
Looking good, just need some more tests

File: src/app/devices/event-log/event-log.component.ts (hunk 1 above)
Comment (chethanbellechalu-intel) [reply src/app/devices/event-log/event-log.component.ts:68]: Added unit test

File: src/app/devices/event-log/event-log.component.ts (hunk 2 above)
Comment (chethanbellechalu-intel) [reply src/app/devices/event-log/event-log.component.ts:45]: Updated API response to cover all the cases

**Review by rsdmike — APPROVED** [review]

**Pull request merged**